### ISSUE TITLE
No such table 

### ISSUE THREAD

**Issue by mangojuice0** [tomaarsen/TwitchMarkovChain#20]
I'm trying to run the bot but it just outputs this

[2021-09-20 01:10:14,417] [Database] [INFO    ] - Updating Database to new version - supports better punctuation handling.
[2021-09-20 01:10:14,423] [Database] [INFO    ] - Created a copy of the database called "MarkovChain_yabbe_modified.db". The update will modify this file.
Traceback (most recent call last):
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\MarkovChainBot.py", line 570, in <module>
    MarkovChain()
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\MarkovChainBot.py", line 30, in __init__ 
    self.db = Database(self.chan)
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\Database.py", line 96, in __init__       
    self.update_v3(channel)
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\Database.py", line 411, in update_v3
    modify_start(table)
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\Database.py", line 331, in modify_start
    data = self.execute(f"SELECT * FROM {table};", fetch=True)
  File "C:\Users\wrk\Desktop\bt\TwitchMarkovChain\Database.py", line 516, in execute
    cur.execute(sql)
sqlite3.OperationalError: no such table: MarkovStartA

What am i missing?

**Comment by tomaarsen** [comment]
@mangojuice0 I suspect you are using an older version with a database format that the bot didn't expect.
It tried to update the database to the newest version, which failed, because the original database didn't contain a `MarkovStartA` table. During the updating process, a new database called "MarkovChain_yabbe_modified.db" will be made, which you can safely delete.

Now, I'm not quite sure which database version you are using. In order to find out, I'd like to ask you to execute the following:
```python
[sic]\TwitchMarkovChain> python
Python 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlite3
>>> conn = sqlite3.connect("MarkovChain_yabbe.db")    
>>> cursor = conn.cursor()
>>> cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")   
<sqlite3.Cursor object at 0x000002409EBB2EA0>
>>> cursor.fetchall()
```
In short, this imports `sqlite3`, which can interact with the database table. Then we create a connection to your database, and we grab a cursor on which we can execute SQL. The SQL query finds all tables, and the last command should output a list of tables. Could you create a comment with this list?

**Comment by mangojuice0** [comment]
It works now. What I did is I downloaded the 1.0 version and made it generate a MarkovChain.db file which i then copied to the 2.1 version. It then automatically updated the database to the newer version.

**Comment by tomaarsen** [comment]
Wonderful! Glad to hear that it works now.
